packages/@react-aria/focus/src/FocusScope.tsx

```diff
@@ -470,11 +493,29 @@ function useRestoreFocus(scopeRef: RefObject<Element[]>, restoreFocus: boolean,
         document.removeEventListener('keydown', onKeyDown, true);
       }
 
-      if (restoreFocus && nodeToRestore && isElementInScope(document.activeElement, scopeRef.current)) {
+      // if we already lost focus to the body and this was the active scope, then we should attempt to restore
+      if (
+        restoreFocus
+        && nodeToRestore
+        && (
+          isElementInScope(document.activeElement, scopeRef.current)
+          || (document.activeElement === document.body && activeScope === scopeRef)
```

Comment:
If focus was lost to the body, wouldn't (shouldn't?) activeScope already be reset to `null`?

Reply:
We don't set active scope when we lose focus to the body, we could start tracking that though, but it might be complex because it'd really just be saying that the root is the active scope for anything not in an actual FocusScope, which is different than saying it's on document.body

Reply:
that's a great point, this is going to be a weird race potentially, we should focus on this when we talk this afternoon